simple_salesforce/login.py

```diff
@@ -34,19 +36,28 @@ def SalesforceLogin(
     * security_token -- the security token for the username
     * organizationId -- the ID of your organization
             NOTE: security_token an organizationId are mutually exclusive
-    * sandbox -- True if you want to login to `test.salesforce.com`, False if
-                 you want to login to `login.salesforce.com`.
+    * sandbox -- DEPRECATED: Use login_domain instead.
     * sf_version -- the version of the Salesforce API to use, for example
                     "27.0"
     * proxies -- the optional map of scheme to proxy server
     * session -- Custom requests session, created in calling code. This
                  enables the use of requets Session features not otherwise
                  exposed by simple_salesforce.
     * client_id -- the ID of this client
+    * login_domain -- The domain to using for logging in. Use common login
+                      domains, such as 'login' or 'test', or named domain.
     """
-
-    soap_url = 'https://{domain}.salesforce.com/services/Soap/u/{sf_version}'
-    domain = 'test' if sandbox else 'login'
+    if 'sandbox' in kwargs.keys():
```

Comment:
Perhaps add an assertion here ensuring both parameters aren't being used? Given the logic below, that'd lead to unexpected results.

Reply:
The assertion wouldn't really work if `sandbox` is expected to be True or False (translating to 'login' or 'test'). I can either default to keyword to None, and then assert treat True/False as being used. Otherwise, we'd need to come up with some other assertion logic.

Reply:
I went ahead and set sandbox=None and domain=None, and then updated the logic appropriately. This is the only way I could think to fully allow both arguments but still regulate the expected behavior.